DataFormats/L1TrackTrigger/interface/TTTrack.h

```diff
@@ -296,96 +280,119 @@ void TTTrack<T>::setFitParNo(unsigned int nPar) {
 
 template <typename T>
 GlobalVector TTTrack<T>::momentum() const {
-  return theMomentum;
+  return theMomentum_;
 }
 
 template <typename T>
 double TTTrack<T>::rInv() const {
-  return theRInv;
+  return theRInv_;
 }
 
 template <typename T>
 double TTTrack<T>::tanL() const {
-  return theTanL;
+  return theTanL_;
 }
 
 template <typename T>
 double TTTrack<T>::eta() const {
-  return theMomentum.eta();
+  return theMomentum_.eta();
 }
 
 template <typename T>
 double TTTrack<T>::phi() const {
-  return thePhi;
+  return thePhi_;
 }
 
 template <typename T>
 double TTTrack<T>::d0() const {
-  return theD0;
+  return theD0_;
 }
 
 template <typename T>
 double TTTrack<T>::z0() const {
-  return theZ0;
+  return theZ0_;
 }
 
 template <typename T>
 GlobalPoint TTTrack<T>::POCA() const {
-  return thePOCA;
+  return thePOCA_;
 }
 
 /// Chi2
 template <typename T>
 double TTTrack<T>::chi2() const {
-  return theChi2;
+  return theChi2_;
+}
+
+/// Chi2Z
+template <typename T>
+double TTTrack<T>::chi2Z() const {
+  return theChi2_Z_;
 }
 
-/// Chi2z
+/// Chi2XY
 template <typename T>
-double TTTrack<T>::chi2z() const {
-  return theChi2Z;
+double TTTrack<T>::chi2XY() const {
+  return theChi2_XY_;
 }
 
 /// Chi2 reduced
 template <typename T>
 double TTTrack<T>::chi2Red() const {
-  return theChi2 / (2 * theStubRefs.size() - numFitPars);
+  return theChi2_ / (2 * theStubRefs.size() - theNumFitPars_);
 }
 
 /// MVA quality variables
 template <typename T>
 double TTTrack<T>::trkMVA1() const {
-  return theTrkMVA1;
+  return theTrkMVA1_;
 }
 
 template <typename T>
 double TTTrack<T>::trkMVA2() const {
-  return theTrkMVA2;
+  return theTrkMVA2_;
 }
 
 template <typename T>
 double TTTrack<T>::trkMVA3() const {
-  return theTrkMVA3;
+  return theTrkMVA3_;
 }
 
 /// StubPtConsistency
 template <typename T>
 void TTTrack<T>::setStubPtConsistency(double aStubPtConsistency) {
-  theStubPtConsistency = aStubPtConsistency;
+  theStubPtConsistency_ = aStubPtConsistency;
   return;
 }
 
 /// StubPtConsistency
 template <typename T>
 double TTTrack<T>::stubPtConsistency() const {
-  return theStubPtConsistency;
+  return theStubPtConsistency_;
 }
 
+/// StubPtConsistency
+template <typename T>
+unsigned int TTTrack<T>::hitPattern() const {
+  return theHitPattern_;
+}
+
+/// StubPtConsistency
```

Comment:
Same here, would be good to correct comment to avoid confusion.

Reply:
Agreed. I have a few further comments:

1) The digitisation ranges assumed at https://github.com/mdhildreth/cmssw/blob/uptdateTTTrackForChi2Split/DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h#L219 appear wildly inconsistent with those specified in https://twiki.cern.ch/twiki/bin/view/CMS/HybridDataFormat#Fitted_Tracks_written_by_KalmanF . Please check.

2) Please add two extra functions to TTTrack.h:

chi2XYred(){return theChi2XY_/(theStubRefs.size() - (theNumFitPars_-2));}
chi2Zred(){return theChi2Z_/(theStubRefs.size() - 2);}

3) If you have adopted the convention that class data members are distinguished by naming them theXYZ_, then variables such as https://github.com/mdhildreth/cmssw/blob/uptdateTTTrackForChi2Split/DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h#L26 should not include "the"

4) Please eliminate the hard-wired constants, such as 16 at https://github.com/mdhildreth/cmssw/blob/uptdateTTTrackForChi2Split/DataFormats/L1TrackTrigger/src/TTTrack_TrackWord.cc#L167 . This could be written instead (NZ0Bits+1)+NChi2Bits .

Cheers,
Ian

Reply:
**Responding to comments:**

The last commit gets most of these, including Louise's suggestions.

Responding to Ian's comments: @tomalin 

1. Fixed, except for phi0. I assumed that phi would be digitized with respect to the center of a nonant, which gets you maximum resolution (if we could figure out how to calculate it).  I don't know/understand why phi0 should range between +/- 1.026 ?  Why is that the magic number?

2. Done
3. Done
4. Sorry, I don't have time to do this now.  Since the Track Word format is likely to change, I also don't want to spend a whole bunch of time prettying up this code at this point.  I suggest we press on with this version unless it's critical.  
5. Done

Cheers,
Mike

Reply:
1) The digitisation range in phi0 of +-1.026 rads about the centre of the sector. This is because the required stub phi digi range is 2*pi/9 + 2*hourglass_overlap (2*0.1638) = 1.026; i.e. from -1.026/2 to +1.026/2. To simplify maths, the phi0 digi range can differ from this only by powers of 2. And unfortunately, this range is very slightly too small to cover the allowed phi0 values of tracks in a nonant, so we have to double it to (-1.026,+1.026).
2) I just noticed that TTTrackWord is digitising eta instead of tanL. This is also inconsistent with the agreement documented on the twiki.

Reply:
@tomalin @skinnari
Ok, I have fixed the last two issues with the phi digitization and switched eta -> tanl in the Track Word.   I guess someone downstream will do the conversion from tanl back to eta so it can be used in the trigger.

Reply:
My comment (2) above was incorrectly implemented in the C++. The number of degrees of freedom of the chi2XY should be "(theStubRefs.size() - (theNumFitPars_-2))", but at https://github.com/cms-sw/cmssw/blob/9deefb0d87de9ad4a85b9c12cf28723725636493/DataFormats/L1TrackTrigger/interface/TTTrack.h#L347 it is instead given as "(theStubRefs.size() - theNumFitPars_ - 2)". 

N.B. As sanity check, the number of degrees of freedom in XY plus the number in Z should equal the number in total, which is not currently the case in the C++.

Reply:
fixed